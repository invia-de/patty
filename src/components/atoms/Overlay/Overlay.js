import React from 'react';
import PropTypes from 'prop-types';
import cx from '../../../utils/classnames';

import styles from './Overlay.module.scss';

/**
 * @author [Roman Semko](mailto:roman.semko-extern@invia.de)
 * @since 0.1.0
 */
export default function Overlay({
  children,
  className,
  open,
  onClick,
  isStatic
}) {
  if (!open) {
    return null;
  }
  return (
    <div
      onClick={onClick}
      className={cx(styles.overlay, isStatic && styles.static, className)}
    >
      {children}
    </div>
  );
}

Overlay.propTypes = {
  /** additional classNames you want to add */
  className: PropTypes.string,
  /** required due to accessibility */
  children: PropTypes.node.isRequired,
  open: PropTypes.bool,
  /** When the overly gets clicked */
  onClick: PropTypes.func,
  /* prevents modal from unfolding full-screen on smaller devices */
  isStatic: PropTypes.bool
};
