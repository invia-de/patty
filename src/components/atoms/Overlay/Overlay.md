Add `open` to `<Overlay open>` to display the overlay!

```jsx
<Overlay>
  <div
    style={{
      width: 468,
      height: 40,
      backgroundColor: 'white',
      textAlign: 'center',
      paddingTop: 20
    }}
  >
    Overlay
  </div>
</Overlay>
```
