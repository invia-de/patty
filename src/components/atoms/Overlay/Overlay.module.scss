@import 'theme.scss';

$overlay_background-color: rgba(0, 0, 0, 0.3) !default;
$overlay_padding: 10px !default;

.overlay {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: $overlay_background-color;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;

  .content {
    padding: $overlay_padding;
  }

  @include respond-below(sm) {
    overflow: auto;
    display: block;
  }
}
