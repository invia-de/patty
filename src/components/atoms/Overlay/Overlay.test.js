import React from 'react';
import Overlay from './Overlay';
import { render } from '@testing-library/react';

test('Closed overlay renders', () => {
  const overlay = (
    <Overlay>
      <div
        style={{
          width: 468,
          height: 40,
          backgroundColor: 'white',
          textAlign: 'center',
          paddingTop: 20
        }}
      >
        Overlay
      </div>
    </Overlay>
  );
  const node = render(overlay);
  expect(node.container).toMatchSnapshot();
  expect(node.container.firstChild).toBeNull();
});

test('Open overlay renders', () => {
  const overlay = (
    <Overlay open>
      <div
        style={{
          width: 468,
          height: 40,
          backgroundColor: 'white',
          textAlign: 'center',
          paddingTop: 20
        }}
      >
        Overlay
      </div>
    </Overlay>
  );
  const node = render(overlay);

  expect(node.container).toMatchSnapshot();
  expect(node.container.firstChild).toBeTruthy();
  expect(node.container.firstChild.children[0].textContent).toEqual('Overlay');
});

test('Overlay click works', () => {
  const onClick = jest.fn();
  const overlay = (
    <Overlay open onClick={onClick}>
      <div
        style={{
          width: 468,
          height: 40,
          backgroundColor: 'white',
          textAlign: 'center',
          paddingTop: 20
        }}
      >
        Overlay
      </div>
    </Overlay>
  );
  const node = render(overlay);

  node.container.firstChild.click();
  expect(onClick).toHaveBeenCalled();
});
