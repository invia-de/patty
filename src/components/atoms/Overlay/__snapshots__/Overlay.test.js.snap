// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Closed verlay renders 1`] = `<div />`;

exports[`Open verlay renders 1`] = `
<div>
  <div
    class="overlay"
  >
    <div
      style="width: 468px; height: 40px; background-color: white; text-align: center; padding-top: 20px;"
    >
      Overlay
    </div>
  </div>
</div>
`;
